memory_ttl.go

```diff
@@ -41,23 +44,48 @@ func NewMemoryWithTTL(ttl time.Duration) *MemoryTTL {
 
 // StartGC starts the garbage collection process in a go routine
 func (r *MemoryTTL) StartGC(gcInterval time.Duration) {
-	r.gcInterval = gcInterval
+	if gcInterval <= 0 {
+		return
+	}
+
+	r.gcTicker = time.NewTicker(gcInterval)
+	r.done = make(chan struct{})
+
 	go func() {
-		for _ = range time.Tick(gcInterval) {
-			for key := range r.cache.items {
-				if !r.isValid(key) {
-					r.Delete(key)
+		for {
+			select {
+			case <-r.gcTicker.C:
+				now := time.Now()
+
+				r.Lock()
```

Comment:
could you explain what you mean by starvation and how could that affect other operations in this context?